src/ophyd_async/epics/motion/motor.py

```diff
@@ -43,13 +121,83 @@ def __init__(self, prefix: str, name="") -> None:
         super().__init__(name=name)
 
     def set_name(self, name: str):
         super().set_name(name)
         # Readback should be named the same as its parent in read()
         self.user_readback.set_name(name)
 
+    @AsyncStatus.wrap
+    async def prepare(self, value: FlyMotorInfo):
+        """Calculate required velocity and run-up distance, then if motor limits aren't
+        breached, move to start position minus run-up distance"""
+
+        self.fly_info = value
+
+        # Velocity at which motor travels from start_position to end_position, in mm/s
+        fly_velocity = await self._prepare_velocity()
+
+        await self._prepare_motor_path(fly_velocity)
+        await self.set(self.fly_info.prepared_position)
+
+    async def kickoff(self):
+        """Begin moving motor from prepared position to final position. Mark as
+        complete once motor reaches start position"""
+        assert self.fly_info, "Motor must be prepared before attempting to kickoff"
+
+        async def has_motor_kicked_off():
+            assert self.fly_info
+
+            if self.fly_info.direction:
+                async for value in observe_value(self.user_readback, timeout=60):
+                    if value >= self.fly_info.prepared_position:
+                        self.fly_info.complete_time_elapsed = time()
+                        break
+            else:
+                async for value in observe_value(self.user_readback, timeout=60):
+                    if value <= self.fly_info.prepared_position:
+                        self.fly_info.complete_time_elapsed = time()
+                        break
+
+        await self.set(self.fly_info.completed_position)
+
+        # Kickoff complete when motor reaches start position
+        return AsyncStatus(has_motor_kicked_off())
+
+    @WatchableAsyncStatus.wrap
+    async def complete(self):
+        """Send periodic updates of the motor's fly progress. Mark as complete once
+        motor reaches end position."""
+
+        assert (
+            self.fly_info
+        ), "Motor must be prepared and kicked off before attempting to complete"
+
+        units, precision = await asyncio.gather(
+            self.motor_egu.get_value(),
+            self.precision.get_value(),
+        )
+
+        async for value in observe_value(
+            self.user_readback, timeout=self.fly_info.complete_timeout
+        ):
+            yield WatcherUpdate(
+                name=self.name,
+                current=value,
+                initial=self.fly_info.prepared_position,
+                target=self.fly_info.end_position,
+                unit=units,
+                precision=precision,
+                time_elapsed=round(time() - self.fly_info.complete_time_elapsed, 2),
+                time_remaining=self.fly_info.time_for_move
+                - self.fly_info.complete_time_elapsed,
+            )
+
+            if value >= self.fly_info.end_position:
+                break
```

Comment:
I don't think we care that it has actually started, just that we have kicked it off, so this can be simplified to:
```suggestion
    @AsyncStatus.wrap
    async def kickoff(self):
        """Begin moving motor from prepared position to final position. Mark as
        complete once motor reaches start position"""
        assert self._completed_position, "Motor must be prepared before attempting to kickoff"
        self._fly_status = self.set(self._completed_position)

    def complete(self) -> WatchableAsyncStatus:
        """Send periodic updates of the motor's fly progress. Mark as complete once
        motor reaches end position."""
        assert self._fly_status, "kickoff not called"
        return self._fly_status
```

Reply:
Looks good - this also means `complete` will be done once the motor has travelled to `end_position` + `run_up_distance`. Is that fine? The docstrings for `kickoff` and `complete` are written with respect to acquisition, so is a bit ambiguous to what it should mean for motors